sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_internal/symmetric_key.py

```diff
@@ -94,15 +101,17 @@ def default_encryption_algorithm(self):
     def default_key_wrap_algorithm(self):
         return _default_kw_alg_by_size[len(self._key)]
 
-    def encrypt(self, plain_text, iv, **kwargs):  # pylint:disable=arguments-differ
+    def encrypt(self, plain_text, **kwargs):  # pylint:disable=arguments-differ
         algorithm = self._get_algorithm("encrypt", **kwargs)
-        encryptor = algorithm.create_encryptor(key=self._key, iv=iv)
-        return encryptor.transform(plain_text, **kwargs)
+        raise_if_incorrect_key_size(algorithm, len(self._key))
+        encryptor = algorithm.create_encryptor(key=self._key, iv=kwargs.pop("iv", None))
```

Comment:
Won't this just raise when `iv` is None?

Reply:
An error will raise from a check in `aes_cbc.py`, yes. I didn't enforce that check here partially because there are some symmetric algorithms (AES-GCM) that don't accept an IV for encryption. This could be a case of planning for functionality we don't yet have, but I think it makes sense to validate that while validating the key and IV size. Is that what you were asking about?

Reply:
I thought it was odd to use a default value that always gets an error. If AES-GCM is the only such algorithm supported by Key Vault, I'd forego the flexibility because we won't ever do AES-GCM locally.